Amazon.QLDB.Driver/driver/AsyncQldbDriver.cs

```diff
@@ -103,5 +230,51 @@ public async Task<IEnumerable<string>> ListTableNames(CancellationToken cancella
 
             return (await result.ToListAsync(cancellationToken)).Select(i => i.StringValue);
         }
+
+        private async Task<AsyncQldbSession> GetSession()
+        {
+            this.logger.LogDebug(
+                "Getting session. There are {} free sessions and {} available permits.",
+                this.sessionPool.Count,
+                this.sessionPool.BoundedCapacity - this.poolPermits.CurrentCount);
+
+            if (await this.poolPermits.WaitAsync(DefaultTimeoutInMs))
+            {
+                var session = this.sessionPool.Count > 0 ? this.sessionPool.Take() : null;
```

Comment:
Seems a bit more complicated than necessary, why not in the if block check if count == 0 and then start a new session, otherwise return sessionPool.Take() ?

Reply:
Good catch. Was a simple copypaste port so I didnt think about it.